src/Sylius/Behat/Client/Request.php

```diff
@@ -0,0 +1,148 @@
+    }
+
+    public static function subResourceIndex(string $resource, string $id, string $subResource, string $token): self
+    {
+        return new self(
+            sprintf('/new-api/%s/%s/%s', $resource, $id, $subResource),
+            HttpRequest::METHOD_GET,
+            ['HTTP_Authorization' => 'Bearer '.$token]
+        );
+    }
+
+    public static function show(string $resource, string $id, string $token): self
+    {
+        return new self(
+            sprintf('/new-api/%s/%s', $resource, $id),
+            HttpRequest::METHOD_GET,
+            ['HTTP_Authorization' => 'Bearer '.$token]
+        );
+    }
+
+    public static function create(string $resource, string $token): self
+    {
+        return new self(
+            '/new-api/'.$resource,
+            HttpRequest::METHOD_POST,
+            ['CONTENT_TYPE' => 'application/json', 'HTTP_Authorization' => 'Bearer '.$token]
+        );
+    }
+
+    public static function update(string $resource, string $id, string $token): self
+    {
+        return new self(
+            sprintf('/new-api/%s/%s', $resource, $id),
+            HttpRequest::METHOD_PUT,
+            ['CONTENT_TYPE' => 'application/ld+json', 'HTTP_Authorization' => 'Bearer '.$token]
+        );
+    }
+
+    public static function delete(string $resource, string $id, string $token): self
+    {
+        return new self(
+            sprintf('/new-api/%s/%s', $resource, $id),
+            HttpRequest::METHOD_DELETE,
+            ['HTTP_Authorization' => 'Bearer '.$token]
+        );
+    }
+
+    public static function transition(string $resource, string $id, string $transition, string $token): self
+    {
+        return new self(
+            sprintf('/new-api/%s/%s/%s', $resource, $id, $transition),
+            HttpRequest::METHOD_PATCH,
+            ['CONTENT_TYPE' => 'application/merge-patch+json', 'HTTP_Authorization' => 'Bearer '.$token]
+        );
+    }
+
+    public static function custom(string $url, string $method, string $token): self
+    {
+        return new self($url, $method, ['HTTP_Authorization' => 'Bearer '.$token]);
+    }
+
+    public function url(): string
+    {
+        return $this->url;
+    }
+
+    public function method(): string
+    {
+        return $this->method;
+    }
+
+    public function headers(): array
+    {
+        return $this->headers;
+    }
+
+    public function content(): string
+    {
+        return json_encode($this->content);
+    }
+
+    public function setContent(array $content): void
+    {
+        $this->content = $content;
+    }
+
+    public function updateContent(array $newValues): void
+    {
+        $this->content = $this->mergeArraysUniquely($this->content, $newValues);
+    }
+
+    public function addSubResource(string $key, array $subresource): void
+    {
+        $this->content[$key][] = $subresource;
+    }
+
+    private function mergeArraysUniquely(array $firstArray, array $secondArray): array
```

Comment:
What's the difference from `array_merge_recursive`? Would be cool to mention.

Reply:
It does not duplicate values for the same keys. But yes, it should be in the comment 👌